source/routing/next_stop_time.cpp

```diff
@@ -338,6 +341,126 @@ NextStopTime::tardiest_stop_time(const StopEvent stop_event,
     return first_discrete_st_pair;
 }
 
+
+template<typename F>
+static void vj_loop(const nt::DiscreteVehicleJourney*, F f) {
+    // Discrete Vehicle Journey doesn't need to loop
+    f(0);
+}
+
+template<typename F>
+static void vj_loop(const nt::FrequencyVehicleJourney* vj, F f) {
+    for (auto freq_shift = vj->start_time; freq_shift <= vj->end_time; freq_shift+= vj->headway_secs) {
+        f(freq_shift);
+    }
+}
+
+using DtSt = std::pair<DateTime, const type::StopTime*>;
+template<typename VJ_T>
+static void fill_cache(const DateTime from,
+        const DateTime to,
+        const type::RTLevel rt_level,
+        const type::AccessibiliteParams& accessibilite_params,
+        const JourneyPattern& jp,
+        const std::vector<const VJ_T*>& vjs,
+        IdxMap<JourneyPatternPoint, std::vector<DtSt>>& arrival,
```

Comment:
rename those params to know that it's "returned" : `arrival_cache` (or better if you find something else)